test/mongodb.test.js

```diff
@@ -547,6 +687,200 @@ describe('mongodb connector', function() {
+              'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+              'testDecimal': '99.55',
+            },
+            {
+              'ObjInsideAnArray': null,
+              'nestedArray': null,
+              'testNumber': '5678',
+              'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+              'testDecimal': '99.55',
+            },
+          ],
+          'innerObj': {
+            'innerObj': {
+              'innerObj': {
+                'innerObj': {
+                  'testNumber': '6666',
+                  'testDate': '2019-01-30 18:26:38.551022',
+                  'testDecimal': '55.55',
+                },
+              },
+            },
+          },
+          'duration': '135',
+          'rating': '4.5',
+        },
+      }, function(err, inst) {
+        // inst.toObject().awards.prizeMoney.should.equal(Decimal128.fromString('25000.00'));
+        var updateData = {
+          'imdb': {
+            'reviewDate': '2019-01-29T04:00:39.828Z',
+            'innerArray': [
+              {
+                'ObjInsideAnArray': {
+                  'testNumber': '7777',
+                  'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                  'testDecimal': '77.77',
+                },
+                'nestedArray': [
+                  {
+                    'testNumber': '7777',
+                    'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                    'testDecimal': '77.77',
+                  },
+                  {
+                    'testNumber': '7777',
+                    'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                    'testDecimal': '77.77',
+                  },
+                ],
+                'testNumber': '7777',
+                'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                'testDecimal': '77.77',
+              },
+              {
+                'ObjInsideAnArray': {
+                  'testNumber': '7777',
+                  'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                  'testDecimal': '77.77',
+                },
+                'nestedArray': null,
+                'testNumber': '7777',
+                'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                'testDecimal': '77.77',
+              },
+            ],
+            'innerObj': {
+              'innerObj': {
+                'innerObj': {
+                  'innerObj': {
+                    'testNumber': '7777',
+                    'testDate': '2019-01-30 18:26:38.551022',
+                    'testDecimal': '77.77',
+                  },
+                },
+              },
+            },
+            'duration': '135',
+            'rating': '7.5',
+          },
+        };
+        modelWithDeepNestedDecimalProps.updateAll({id: inst.id}, updateData, function(err, inst) {
+          inst.toObject().imdb.rating.should.equal(Decimal128.fromString('7.5'));
+        });
```

Comment:
do you want to test 'testDecimal' in the super nested `innerObj` property as well(since you defined it in the model)?